### ISSUE TITLE
Provide ability to use specific ip:port for Foxx applications instead of the standard API port

### ISSUE THREAD

**Issue by frankmayer** [arangodb/arangodb#855]
While it is possible to do port forwarding to accomplish this, it would be practical to be able to provide the ability to define an ip address and/or port for a FOXX application to listen to.

It is good to have this, in case one builds an API in ArangoDB for external users to directly connect to. 
It's better to use standard ports like 80/443 instead of 8529, as non-standard ports are sometimes blocked by proxies.

Also, it would be cleaner to have a separation of such external facing API's and the standard ArangoDB port 8529 API for management purposes.

What do you guys think?

**Comment by pluma** [comment]
I'm closing this issue because we have decided that this use case is out of scope for the project itself. There are plenty of good reverse proxy solutions that provide this capability, as described [in this cookbook recipe](https://docs.arangodb.com/cookbook/MakingFoxxAppAccessible.html).

ArangoDB 3.0 will further improve support for reverse proxies by supporting the `X-Forwarded-*` series of headers (which is opt-in for security reasons), i.e. if the service is configured to trust proxies, it will transparently apply the `X-Forwarded-Host` header instead of the `Host` header to derive `req.hostname`: https://github.com/arangodb/arangodb/blob/devel/js/server/modules/%40arangodb/foxx/router/request.js#L198

You can find further information on configuring Apache and nginx as reverse proxies at e.g. http://citadel.org/doku.php/faq:installation:apacheproxy
